src/panels/ItemPanel.tsx

```diff
@@ -174,37 +154,28 @@ export default class ItemPanel extends React.Component<Props, State> {
 								modalRemoveItem: true
 							})
 						}}
-						checkboxChecked={this.state.selectedItems.includes(id)}
-						handleShowCheckbox={() => {
-							let items = this.state.selectedItems
-							if (items.includes(id)) {
-								items.splice(items.indexOf(id), 1)
-							} else {
-								items.push(id);
-							}
-							this.setState({selectedItems: items, selectionMode: (items.length > 0)})
+						handleSelect={() => {
+							if (ProjectSettings.selectedElements.includes(id))
+								unhighlightElement(id)
+							else highlightElement(id);
+							this.forceUpdate();
 						}}
-						showCheckbox={this.state.selectionMode}
-						selectedItems={this.state.selectedItems}
 						clearSelection={() => {
-							this.setState({selectedItems: [], selectionMode: false})
+							resetDiagramSelection();
+							this.forceUpdate();
 						}}
 						showDetails={() => {
-							unHighlightAll();
-							ProjectSettings.selectedCells = [];
+							highlightElement(id);
 							this.props.updateDetailPanel(id);
 							let elem = graph.getElements().find(elem => elem.id === id);
 							if (elem) {
 								const scale = paper.scale().sx;
 								paper.translate(0, 0);
 								paper.translate((-elem.position().x * scale) + (paper.getComputedSize().width / 2) - elem.getBBox().width,
 									(-elem.position().y * scale) + (paper.getComputedSize().height / 2) - elem.getBBox().height);
-								Diagrams[ProjectSettings.selectedDiagram].origin = {
-									x: paper.translate().tx,
-									y: paper.translate().ty
-								};
-								Diagrams[ProjectSettings.selectedDiagram].scale = paper.scale().sx;
+								updateDiagramPosition(ProjectSettings.selectedDiagram);
 							}
+							this.forceUpdate();
```

Comment:
That's a lot of forcing of updates. Apps are nowdays usually written the other way around - components listen to data / model changes (using e.g. React state, Redux or Observables).

Reply:
Agreed... Redux would be perfect for this IMO, but such a refactor would be in a PR of its own.

Reply:
I am not so sure about Redux TBH, the Observables may be a better fit. The problem with Redux is that you can store only basic values (i.e. simple objects), whereas with Observables it's OK to manage classes / instances. In addition it can be decoupled better from React and you can also use it to trigger / force any component update if necessary.

The biggest issue I see with Redux is performance though, e.g. if a diagram element mouse move triggered a store update, the performance would be sluggish at best, even with some optimizations in place.